src/main/java/calculator/operation/ArithmeticOperation.java

```diff
@@ -0,0 +1,41 @@
+package calculator.operation;
+
+import calculator.exception.ErrorMessage;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+public enum ArithmeticOperation {
+
+    PLUS("+", Math::addExact),
+    MINUS("-", Math::subtractExact),
+    TIMES("*", Math::multiplyExact),
+    DIVIDE("/", Math::floorDiv);
+
+    private final String expression;
+    private final BiFunction<Integer, Integer, Integer> function;
+
+    public static final Map<String, ArithmeticOperation> map = new HashMap<>();
+    static {
+        Arrays.stream(ArithmeticOperation.values())
+                .forEach(value -> map.put(value.expression, value));
+    }
```

Comment:
💯 
아주 약간만 더 손보자면 아래처럼 구현할 수 있어요
```suggestion
    public static final Map<String, ArithmeticOperation> OPERATIONS =
              Arrays.stream(ArithmeticOperation.values())
                         .collect(Collectors.toMap(ArithmeticOperation::getExpression, Function.identity()));
    
    private String getExpression() { return expression; }
```
잘 하셨어요😄

Reply:
아차 그리고 외부에서는 사용할일이 없으니 `private` access modifier를 사용하는게 좋겠네요

Reply:
생성과 동시에 초기화 하는 방법이 좋을거 같네요! ㅎㅎ

```Collectors.toMap(ArithmeticOperation::getExpression, Function.identity())```
- 단지 람다식을 이용하기 위해 private getter method 를 추가할 필요가 있을까 하는 궁금증이 생기네요 ❓ 
- 람다식의 장점과 어떤 경우에는 stream 대신 단순 반복문을 사용하는 것이 효율적인지에 대해서도 알아봐야 할 거 같아요!

Reply:
음 `getter`라고 하더라도 `private`이기 때문에 큰 문제 없지 않을까요??